conv/abci/block.go

```diff
@@ -38,3 +38,49 @@ func ToABCIHeader(header *types.Header) (tmproto.Header, error) {
 		ProposerAddress:    header.ProposerAddress,
 	}, nil
 }
+
+func ToABCIBlock(block *types.Block) (tmproto.Block, error) {
+	abciHeader, err := ToABCIHeader(&block.Header)
+	if err != nil {
+		return tmproto.Block{}, err
+	}
+	abciCommit := ToABCICommit(&block.LastCommit)
+	// This assumes that we have only one signature
+	if len(abciCommit.Signatures) == 1 {
+		abciCommit.Signatures[0].ValidatorAddress = block.Header.ProposerAddress
+	}
+	abciBlock := tmproto.Block{
+		Header: abciHeader,
+		Evidence: tmproto.EvidenceList{
+			Evidence: nil,
+		},
+		LastCommit: abciCommit,
+	}
+	abciBlock.Data.Txs = make([][]byte, len(block.Data.Txs))
+	for i := range block.Data.Txs {
+		abciBlock.Data.Txs[i] = block.Data.Txs[i]
+	}
+
+	return abciBlock, nil
+}
+
+func ToABCICommit(commit *types.Commit) *tmproto.Commit {
+	tmCommit := tmproto.Commit{
+		Height: int64(commit.Height),
+		Round:  0,
+		BlockID:    tmproto.BlockID{
+			Hash:          commit.HeaderHash[:],
+			PartSetHeader: tmproto.PartSetHeader{},
+		},
+	}
+	for _, sig := range commit.Signatures {
+		commitSig := tmproto.CommitSig{
+			BlockIdFlag:      tmproto.BlockIDFlagCommit,
+			//ValidatorAddress: nil,
+			//Timestamp:        nil,
```

Comment:
What are the ramifications of commenting this out vs explicitly setting `nil`?

Reply:
you can't set nil, as its `string` and `time.Time` ;)
TBH it's a dirty work-in-progress version, and I'm reworking this part anyways. I'll move changes to separate branch/PR.

Reply:
Changes moved to #187.